runtime/resolvers/metricsview_time_range.go

```diff
@@ -0,0 +1,129 @@
+package resolvers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/pkg/mapstructureutil"
+	"github.com/rilldata/rill/runtime/queries"
+)
+
+func init() {
+	runtime.RegisterResolverInitializer("metricsview_time_range", newMetricsViewTimeRangeResolver)
+}
+
+type metricsViewTimeRangeResolver struct {
+	runtime    *runtime.Runtime
+	instanceID string
+	query      *queries.MetricsViewTimeRange
```

Comment:
It should not use the `queries` package, the goal of the `resolvers` package is to deprecate the `queries` package altogether. It would be better to copy the logic over here, or ideally implement it as a function on `metricsview.Executor` which is called here.

I know it's tricky to navigate partially refactored code, but since it's not possible to justify doing these refactors in one big push, we need to gradually make them as we require new functionality. The place we want to get to is:
- the `resolvers` package handles all data querying/exporting use cases, with consistent support for APIs, reports, alerts, etc.
- the `metricsview` package implements all metrics-related query generation functionality